### ISSUE TITLE
model.summary() has inconsistent output for identical model (sequential vs functional api)

### ISSUE THREAD

**Issue by JulianCologne** [keras-team/keras#15065]
### Problem

result of `model.summay()` is inconsistent for "identical" models using the sequential and the functional API.

### Steps to reproduce

```python
import keras

# Sequential
model = keras.Sequential(name='Sequential model')
model.add(keras.layers.InputLayer(input_shape=(3,), name='InputLayer'))
model.add(keras.layers.Dense(4, name='OutputLayer'))
model.summary()

# Functional
inputs = keras.layers.Input(shape=(3,), name='InputLayer')
outputs = keras.layers.Dense(4, name='OutputLayer')(inputs)
model = keras.Model(inputs=inputs, outputs=outputs, name='Functional model')
model.summary()
```

### Expected behaviour 

Same output for both because they should represent the identical model. I would like to see the input layer in the sequential model as well.

### Actual behaviour

Sequential API output
```
Model: "Sequential model"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
OutputLayer (Dense)          (None, 4)                 16        
=================================================================
Total params: 16
Trainable params: 16
Non-trainable params: 0
```

Functional API output
```
Model: "Functional model"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
InputLayer (InputLayer)      [(None, 3)]               0         <<<<<<<<<<<<<<<<<<<<<<<<<<<<
_________________________________________________________________
OutputLayer (Dense)          (None, 4)                 16        
=================================================================
Total params: 16
Trainable params: 16
Non-trainable params: 0
```

**Comment by rmothukuru** [comment]
@JulianCologne,
This behavior seems to expected from the documentation of [Functional API](https://keras.io/guides/functional_api/) and [Sequential API](https://keras.io/guides/sequential_model/). What do you think?

**Comment by bwolfson97** [comment]
I agree with @JulianCologne. This confused me at first before I learned that `Sequential` models don't show the initial `InputLayer`. We can see that the [source code](https://github.com/tensorflow/tensorflow/blob/a4dfb8d1a71385bd6d122e4f27f86dcebb96712d/tensorflow/python/keras/engine/sequential.py#L144) just hides the initial `InputLayer` for historical reasons:
```py
  def layers(self):
    # Historically, `sequential.layers` only returns layers that were added
    # via `add`, and omits the auto-generated `InputLayer` that comes at the
    # bottom of the stack.
    # `Trackable` manages the `_layers` attributes and does filtering
    # over it.
    layers = super(Sequential, self).layers
    if layers and isinstance(layers[0], input_layer.InputLayer):
      return layers[1:]
    return layers[:]
```

I think things would be clearer and more consistent if the `InputLayer` was shown in `Sequential` models.

**Comment by gowthamkpr** [comment]
@JulianCologne This is an expected behavior and is not a bug of any sort. Can you please go ahead and close the issue. Thanks!

**Comment by JulianCologne** [comment]
> @JulianCologne This is an expected behavior and is not a bug of any sort. Can you please go ahead and close the issue. Thanks!

From my point of view as a user this is unexpected and inconsistent behavior like I explained above.

I agree that this is not a bug (I did not add that label) but still a "problem".

I have not heard any reasonable reply why this cannot/shouldn't be changed (other than "Historically it was like this...")? Looks like this can be changed by simply removing 2 lines of code!?
